internal/resources/helmrelease/resourec_helm_release.go

```diff
@@ -371,26 +372,49 @@ func updateCheckForSpec(d *schema.ResourceData, atomicSpec *releaseclustermodel.
-	resp, err := config.TMCConnection.ClusterHelmResourceService.VmwareTanzuManageV1alpha1ClusterHelmResourceServiceList(
-		&helmclustermodel.VmwareTanzuManageV1alpha1ClusterFluxcdHelmListHelmRequestParameters{
-			SearchScope: &helmclustermodel.VmwareTanzuManageV1alpha1ClusterFluxcdHelmSearchScope{
-				ClusterName:           scopedFullnameData.FullnameCluster.ClusterName,
-				ManagementClusterName: scopedFullnameData.FullnameCluster.ManagementClusterName,
-				ProvisionerName:       scopedFullnameData.FullnameCluster.ProvisionerName,
+	switch scopedFullnameData.Scope {
+	case commonscope.ClusterScope:
+		resp, err := config.TMCConnection.ClusterHelmResourceService.VmwareTanzuManageV1alpha1ClusterHelmResourceServiceList(
+			&helmclustermodel.VmwareTanzuManageV1alpha1ClusterFluxcdHelmListHelmRequestParameters{
+				SearchScope: &helmclustermodel.VmwareTanzuManageV1alpha1ClusterFluxcdHelmSearchScope{
+					ClusterName:           scopedFullnameData.FullnameCluster.ClusterName,
+					ManagementClusterName: scopedFullnameData.FullnameCluster.ManagementClusterName,
+					ProvisionerName:       scopedFullnameData.FullnameCluster.ProvisionerName,
+				},
 			},
-		},
-	)
+		)
 
-	if err != nil {
-		return err
-	}
+		if err != nil {
+			return err
+		}
 
-	if len(resp.Helms) == 0 {
-		return errors.Errorf("Tanzu mission control helm feature is disable on cluster, name: %s", scopedFullnameData.FullnameCluster.ClusterName)
-	}
+		if len(resp.Helms) == 0 {
+			return errors.Errorf("Tanzu mission control helm feature is disable on cluster, name: %s", scopedFullnameData.FullnameCluster.ClusterName)
+		}
 
-	if _, ok := resp.Helms[0].Status.Conditions[disabledKey]; ok {
-		return errors.Errorf("Tanzu mission control helm feature is disable on cluster, name: %s", scopedFullnameData.FullnameCluster.ClusterName)
+		if _, ok := resp.Helms[0].Status.Conditions[disabledKey]; ok {
+			return errors.Errorf("Tanzu mission control helm feature is disable on cluster, name: %s", scopedFullnameData.FullnameCluster.ClusterName)
+		}
+	case commonscope.ClusterGroupScope:
+		resp, err := config.TMCConnection.ClusterGroupHelmResourceService.VmwareTanzuManageV1alpha1ClustergroupHelmResourceServiceList(
+			&helmclustergroupmodel.VmwareTanzuManageV1alpha1ClustergroupHelmListHelmRequestParameters{
+				SearchScope: &helmclustergroupmodel.VmwareTanzuManageV1alpha1ClustergroupFluxcdHelmSearchScope{
+					ClusterGroupName: scopedFullnameData.FullnameClusterGroup.ClusterGroupName,
+				},
+			},
+		)
+
+		if err != nil {
+			return err
+		}
+
+		if len(resp.Helms) == 0 {
+			return errors.Errorf("Tanzu mission control helm feature is disable on cluster group, name: %s", scopedFullnameData.FullnameCluster.ClusterName)
+		}
+
+		if resp.Helms[0].Status.Phase == nil {
+			return errors.Errorf("Tanzu mission control helm feature is disable on cluster group, name: %s", scopedFullnameData.FullnameCluster.ClusterName)
+		}
 	}
 
 	return nil
```

Comment:
Please fix the file name. Verify other filenames as well.

Reply:
Updated PTAL